File: LeetCode Solutions/January LeetCoding Challenge/Smallest String With A Given Numeric Value.py
class Solution:
    def getSmallestString(self, n: int, k: int) -> str:
        k -= n
        numericValue = '_bcdefghijklmnopqrstuvwxy_'
        lexicographicallySmallestString = 'z' * (k // 25)
        if(k % 25):
            lexicographicallySmallestString = numericValue[k%25] + lexicographicallySmallestString
        return 'a' * (n - len(lexicographicallySmallestString)) + lexicographicallySmallestString

File: LeetCode Solutions/January LeetCoding Challenge/Smallest String With A Given Numeric Value.cpp
class Solution {
public:
    string getSmallestString(int n, int k) {
        k -= n;
        string numericValue = "_bcdefghijklmnopqrstuvwxy_";
        string lexicographicallySmallestString = std::string(~~(k / 25), 'z');
        if(k % 25){
            lexicographicallySmallestString = numericValue[k % 25] + lexicographicallySmallestString;
        }
        return std::string(n - lexicographicallySmallestString.size(), 'a') + lexicographicallySmallestString;
    }
};